### ISSUE TITLE
"no overload variant" with urlparse(..., allow_fragments=...) 

### ISSUE THREAD

**Issue by madeleine-empirical** [python/mypy#2640]
I am confused by an error I'm getting with a urlparse call. This may be related to #1907, but I don't know.

Here's a minimal reproduction case:
```py
from urllib.parse import urlparse

# Gives a mypy error:
u = urlparse('https://example.com/#frag', allow_fragments=False)
print(u)

# Does not give a mypy error:
u2 = urlparse('https://example.com/#frag')
print(u2)
```

If I run this, it gives the output I expect:
```
$ python test.py 
ParseResult(scheme='https', netloc='example.com', path='/#frag', params='', query='', fragment='')
ParseResult(scheme='https', netloc='example.com', path='/', params='', query='', fragment='frag')
```

If I run mypy on it, I get:
```
$ mypy --fast-parser test.py
test.py:4: error: No overload variant of "urlparse" matches argument types [builtins.str, builtins.bool]
$ mypy test.py
test.py:4: error: No overload variant of "urlparse" matches argument types [builtins.str, builtins.bool]
```

The [typeshed code](https://github.com/python/typeshed/blob/master/stdlib/3/urllib/parse.pyi#L123) looks reasonable to me.

software versions:

```sh
$ python --version
Python 3.5.2
$ pip list
mypy-lang (0.4.7.dev0)
pip (9.0.1)
setuptools (32.3.1)
typed-ast (0.6.1)
wheel (0.30.0a0)
```

**Comment by rwbarton** [comment]
The stubs say `allow_framgents`. Oops...

Can you file this on typeshed instead?

**Comment by madeleine-empirical** [comment]
Thanks; that's an easy explanation. PR'd as https://github.com/python/typeshed/pull/812.
